vivisect/tests/testswitches.py

```diff
@@ -0,0 +1,1819 @@
+    ]
+
+
+
+WALKER_MAPS = maps = [\
+    (0x201a5a0, 0x9b, e_mem.MM_RWX, b'UH\x89\xe5H\x83\xec0@\x88\xf0H\x89}\xf8\x88E\xf7H\x89U\xe8H\x8bU\xf8H\x89\xd7H\x83\xc7`H\x8bu\xe8H\x89U\xe0\xe8R\x1a\x00\x00\xa8\x01\x0f\x85\x05\x00\x00\x00\xe9\x05\x00\x00\x00\xe9V\x01\x00\x00H\x8bE\xe0H\x8bH8H\x83\xc18H\x8bu\xe8H\x89\xcf\xe8\x98\x1a\x00\x00H\x89\xc7\xe8\xb0\x1a\x00\x00\x83\xc0\xff\x89\xc1\x83\xe8\x0bH\x89M\xd8\x89E\xd4\x0f\x87\x1c\x01\x00\x00H\x8d\x05\xec\xc6\x05\x00H\x8bM\xd8Hc\x14\x88H\x01\xc2\xff\xe2\x8aE\xf7H\x8bU\xe8H\x8b}\xe0\x0f\xb6\xf0\xe8\x84\x1a\x00'),
+    (0x201a731, 0x1e, e_mem.MM_RWX, b'\xe9\x00\x00\x00\x00H\x83\xc40]\xc3\x0f\x1f@\x00UH\x89\xe5H\x89}\xf8H\x8b}\xf8H\x8b\x07'),
+    (0x2076d08, 0x30, e_mem.MM_READ, b'\x11:\xfa\xff!9\xfa\xff\xe19\xfa\xffi9\xfa\xff\x819\xfa\xff\x999\xfa\xff\xb19\xfa\xff99\xfa\xffQ9\xfa\xff):\xfa\xff\xc99\xfa\xff\xf99\xfa\xff'),
+]
+
+def applyMaps(vw, maps=MS_MAPS, mapbase=0x400000, bufferpages=2):
+    bufferpgsz = bufferpages * 4096
+    vw.addMemoryMap(mapbase - bufferpgsz, e_mem.MM_RWX, 'testswitches', '@' * bufferpgsz)
+
+    mapdelta = mapbase - maps[0][0]
+    
+
+    vw.addFile('testswitches', mapbase, "@@@@@@@@")
+    for mapva, sz, perms, mem in maps:
+        mapva += mapdelta
+        vw.addMemoryMap(mapva, perms, 'testswitches', mem)
+        vw.addSegment(mapva, len(mem), 'switch_code_%x' % mapva, 'testswitches')
+
+
+def genMsSwitchWorkspace(maps=MS_MAPS, mapbase=0x400000, bufferpages=2, arch='amd64'):
+    vw = vivisect.VivWorkspace()
+    vw.setMeta('Architecture', arch)
+    vw.setMeta('Platform', 'windows')
+    vw.setMeta('Format', 'pe')
+    vw._snapInAnalysisModules()
+
+    applyMaps(vw, maps, mapbase, bufferpages)
+    return vw
+
+def genLinuxSwitchWorkspace(maps=LIBC_MAPS, mapbase=0x400000, bufferpages=2):
+    vw = vivisect.VivWorkspace()
+    vw.setMeta('Architecture', 'i386')
+    vw.setMeta('Platform', 'linux')
+    vw.setMeta('Format', 'elf')
+    vw._snapInAnalysisModules()
+
+    applyMaps(vw, maps, mapbase, bufferpages)
+    return vw
+
+
+cbs_ms_0 = [
+    (0x404ee0, 0xd , 0x404ee0),
+    (0x404eed, 0x9 , 0x404ee0),
+    (0x404ef6, 0x9 , 0x404ee0),
+    (0x404eff, 0x26, 0x404ee0),
+    (0x404f25, 0xe , 0x404ee0),
+    (0x404f33, 0xa , 0x404ee0),
+    (0x411232, 0xc , 0x404ee0),
+    (0x41123e, 0x1b, 0x404ee0),
+    (0x411259, 0xa , 0x404ee0),
+    (0x4186af, 0x6 , 0x404ee0),
+    (0x4186b5, 0x9 , 0x404ee0),
+    (0x4186be, 0x9 , 0x404ee0),
+    (0x4186c7, 0x5 , 0x404ee0),
+    (0x4186cc, 0xe , 0x404ee0),
+    (0x419db9, 0xa , 0x404ee0),
+    (0x420d68, 0x11, 0x404ee0),
+    (0x420d79, 0x1a, 0x404ee0),
+    (0x420d93, 0x1a, 0x404ee0),
+    (0x420dad, 0x1a, 0x404ee0),
+    (0x420dc7, 0xf , 0x404ee0)]
+
+cbs_libc_0 = [
```

Comment:
what do you mean?  what does success look like?

Reply:
So other tests separate out code from data so that things can be easily updated. It'd make things easier for when we have to revisit this down the line if we separate those out.